pkg/provisioner/controller.go

```diff
@@ -476,14 +490,18 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 	// Create OB
 	// Note: do not move ob create/update calls before secret.
 	//   spec.Authentication is lost after create/update, which breaks secret creation
-	ob, err = createObjectBucket(
+	// Note: once at this stage, don't allow ob to be nil if returning with error for cleanup
+	// because ob has important information for the provisioner Delete to clean up resources
+	var createdOB *v1alpha1.ObjectBucket
```

Comment:
again, not liking the ob clone here. If `createOB` returned the passed in ob for err!=nil then no info is lost (I think).  What if `createOB` was refactored a bit:
```
func createObjectBucket(ob *v1alpha1.ObjectBucket, c versioned.Interface, retryInterval, retryTimeout time.Duration) (result *v1alpha1.ObjectBucket, err error) {
	logD.Info("creating ObjectBucket", "name", ob.Name)

	err = wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
		result, err = c.ObjectbucketV1alpha1().ObjectBuckets().Create(context.TODO(), ob, metav1.CreateOptions{})
		if errors.IsAlreadyExists(err) {
			err = nil
		} else if err != nil {
			// could be intermittent api error
			log.Error(err, "probably not fatal, retrying")
		}
		return (err == nil), err
	})
	if err == nil {
            return result, nil
       }
      return ob, err
return
}
```

Reply:
I don't think the secret/cm need to be handled today. They aren't used in the cleanup function, and I don't see any places where they could fall to a `nil` state and cause code issues as it stands today.